core/common/src/main/java/alluxio/network/protocol/databuffer/DataBuffer.java

```diff
@@ -40,6 +40,22 @@
   ByteBuffer getReadOnlyByteBuffer();
 
   /**
+   * Transfers this buffer's data to the specified destination starting at
+   * the current {@code readerIndex} and increases the {@code readerIndex}
+   * by the number of the transferred bytes (= {@code length}).
+   *
+   * @param dst the destination
+   * @param dstIndex the first index of the destination
+   * @param length the number of bytes to transfer
+   */
+  void readBytes(byte[] dst, int dstIndex, int length);
```

Comment:
how do you indicate failure for this method? there is no returned value and no exception thrown in this interface